linksharing/handler.go

```diff
@@ -101,20 +94,15 @@ func (handler *Handler) serveHTTP(w http.ResponseWriter, r *http.Request) (err e
 		}
 	}()
 
-	b, err := p.OpenBucket(ctx, bucket, access.EncryptionAccess)
-	if err != nil {
-		handler.handleUplinkErr(w, "open bucket", err)
+	b, err := p.StatBucket(ctx, bucket)
+	if err != nil || b == nil {
+		handler.handleUplinkErr(w, "error finding bucket", err)
 		return err
 	}
-	defer func() {
-		if err := b.Close(); err != nil {
-			handler.log.With(zap.Error(err)).Warn("unable to close bucket")
-		}
-	}()
 
-	o, err := b.OpenObject(ctx, unencPath)
+	o, err := p.DownloadObject(ctx, bucket, key, nil)
```

Comment:
this should just be StatObject, considering the httpranger.ServeContent call doesn't use o.Read at all and only uses o.Info

Reply:
fixed!